extmod/nimble/modbluetooth_nimble.c

```diff
@@ -629,13 +636,12 @@ int mp_bluetooth_init(void) {
 
     // Run the scheduler while we wait for stack startup.
     // On non-ringbuffer builds (NimBLE on STM32/Unix) this will also poll the UART and run the event queue.
-    mp_uint_t timeout_start_ticks_ms = mp_hal_ticks_ms();
+    mp_uint_t start = mp_hal_ticks_ms();
     while (mp_bluetooth_nimble_ble_state != MP_BLUETOOTH_NIMBLE_BLE_STATE_ACTIVE) {
-        uint32_t elapsed = mp_hal_ticks_ms() - timeout_start_ticks_ms;
-        if (elapsed > NIMBLE_STARTUP_TIMEOUT) {
+        if ((mp_hal_ticks_ms() - start) > NIMBLE_STARTUP_TIMEOUT) {
             break;
         }
-        mp_event_wait_ms(NIMBLE_STARTUP_TIMEOUT - elapsed);
+        mp_event_wait_ms(1);
```

Comment:
I'm not sure exactly why this logic was changed?  It should be more efficient to do a sleep for `NIMBLE_STARTUP_TIMEOUT - elapsed` than many sleeps for 1ms.  The idea is that `mp_event_wait_ms()` can return early if there's a system event to process, so it won't wait that whole time if it doesn't need to.

Reply:
I thought this might be over-complicating it for minimal gain, figured it might be better to just simplify it to the same timeout logic used most other places in the codebase. 

But yes it's true the original version should be slightly more efficient with likely no charge in code size so I could change birth these timeouts to the original pattern used here.

Reply:
With the original `mp_event_wait_ms(NIMBLE_STARTUP_TIMEOUT - elapsed);` scheme for both timeouts:
```
   text    data     bss     dec     hex filename
 397964     368   42356  440688   6b970 build-NUCLEO_WB55/firmware.elf
```
With the "simpler" one here `mp_event_wait_ms(1);`:
```
   text    data     bss     dec     hex filename
 397956     368   42356  440680   6b968 build-NUCLEO_WB55/firmware.elf
```

So the calculated timeout costs 4 bytes of flash (per copy of the logic). Considering this is only on boards that already are big enough to hold the BLE stack that's probably reasonable for more efficient use - BLE application are often battery ones and these kind of optimisations do (slightly) improve power use.